tests/e2e/test_rosacli_version.go

```diff
@@ -53,4 +54,46 @@ var _ = Describe("Get CLI version",
 				Expect(stdout).To(ContainSubstring(info.DefaultVersion))
 			},
 		)
+
+		It("list versions can work correctly for hosted-cp cluster via ROSA cli - [id:62088]",
+			labels.Medium,
+			func() {
+				By("Init the client")
+				rosaClient = rosacli.NewClient()
+
+				By("list the versions with --hosted-cp")
+				versionService := rosaClient.Version
+
+				output, err := versionService.ListVersions("", true)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).
+					Should(ContainSubstring(
+						"INFO: Hosted cluster upgrades are cluster-based. To list available upgrades for a cluster, please use 'rosa list upgrades'"))
+
+				By("list the versions with channel group")
+				channelGroups := []string{"stable", "", "candidate", "nightly"}
+				for _, c := range channelGroups {
+					verList, err := versionService.ListAndReflectJsonVersions(c, true)
+					Expect(err).ToNot(HaveOccurred())
+					for _, v := range verList {
+						if !v.Enabled {
+							continue
```

Comment:
Could you confirm if the steps bellow this line in the for loop can be checked? It seems all the below steps are skipped. The v.Enabled is always false as struct OpenShiftVersionTableOutput has no 'Enabled' field

Reply:
I added one print character under the if condition. The below steps are run. v.Enabled is to check the enabled/vaild version. And v struct is OpenShiftVersionJsonOutput.

Reply:
Ok, that's a change included in this PR. sorry to miss that